src/providers/42-school.ts

```diff
@@ -1,5 +1,15 @@
-/** @type {import(".").OAuthProvider} */
-export default function FortyTwo(options) {
+import { OAuthConfig, OAuthUserConfig } from "./oauth"
+
+export interface FortyTwoProfile {
+  id: string
+  fullname: string
+  email: string
+  image: string
```

Comment:
Do you know about any extra fields here?

Reply:
```js
profile: {
    id: 62377,
    email: 'arraji@student.1337.ma',
    login: 'arraji',
    first_name: 'Abdelbari',
    last_name: 'Rraji',
    usual_full_name: 'Abdelbari Rraji',
    usual_first_name: null,
    url: 'https://api.intra.42.fr/v2/users/arraji',
    phone: 'hidden',
    displayname: 'Abdelbari Rraji',
    image_url: 'https://cdn.intra.42.fr/users/arraji.jpg',
    'staff?': false,
    correction_point: 8,
    pool_month: 'august',
    pool_year: '2019',
    location: 'e1r5p15',
    wallet: 195,
    anonymize_date: '2022-11-16T00:00:00.000+01:00',
    created_at: '2019-08-05T10:11:36.151Z',
    updated_at: '2021-11-16T06:51:13.996Z',
    alumni: false,
    'is_launched?': false,
    groups: [],
    cursus_users: [ [Object], [Object] ],
    projects_users: [
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object], [Object], [Object],
      [Object], [Object], [Object]
    ],
    languages_users: [ [Object], [Object] ],
    achievements: [
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object]
    ],
    titles: [],
    titles_users: [],
    partnerships: [],
    patroned: [],
    patroning: [],
    expertises_users: [ [Object], [Object], [Object], [Object], [Object], [Object] ],
    roles: [],
    campus: [ [Object] ],
    campus_users: [ [Object] ],
    user: null
  }
```
I can try to create an interface from the response I get but I can't guarantee the exact value of some field (since it's not documented and I don't have the privilege to edit them to know what they return) so it will be set as `any`

Reply:
sounds good!